src/clarity-angular/select/select.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {ViewChild} from "@angular/core";
+import {
+    AfterViewInit,
+    Component,
+    ContentChildren,
+    ElementRef,
+    EventEmitter,
+    forwardRef,
+    Input,
+    OnDestroy,
+    Optional,
+    Output,
+    QueryList,
+    SkipSelf
+} from "@angular/core";
+import {Subscription} from "rxjs/Subscription";
+
+import {IfOpenService} from "../utils/conditional/if-open.service";
+
+import {State} from "./interfaces/state";
+import {Option} from "./option";
+import {ROOT_SELECT_PROVIDER, RootSelectService} from "./providers/select.service";
+import {SelectInput} from "./select-input";
+import {SelectMenu} from "./select-menu";
+
+@Component({selector: "clr-select", templateUrl: "./select.html", providers: [IfOpenService, ROOT_SELECT_PROVIDER]})
+/**
+ * Basic Version very much WIP
+ */
+export class Select implements OnDestroy {
+    @ContentChildren(Option) options: QueryList<Option>;
+    private model: any;
+    private _loading: boolean;
+    private _clrLoadingText: string;
+    private _subOpen: Subscription;
+    private _subInput: Subscription;
+    private _subSpecialKeys: Subscription;
+    @ViewChild(SelectMenu) selectMenu: ElementRef;
+
+    constructor(public ifOpenService: IfOpenService, public selectService: RootSelectService) {
+        this._subOpen = ifOpenService.openChange.subscribe((value) => {
+            if (value === true) {
+                this.selectService.highlighted = this.options.first;
```

Comment:
had issues with this using lazy loaded options and changed this to nothing select on init. Is it expected behavior that the first one is preselected ? It would be on a normal select but here for now there is now preselection.